src/lab16.py

```diff
@@ -252,14 +259,37 @@ def layout(self):
             child.y = baseline + child.get_ascent()
         max_descent = max([child.get_descent(1.25)
                            for child in self.children])
-        self.height = max_ascent + max_descent
-        self.parent.dirty_height = True
-        mark_dirty(self.parent)
 
-        self.dirty_width = False
-        self.dirty_height = False
-        self.dirty_x = False
-        self.dirty_y = False
+        self.height = self.height_field.set(max_ascent + max_descent)
```

Comment:
Doesn't this mean that height_field should depend on those children? How is that represented?

Reply:
Great great question. Right now I don’t do invalidation for elements of lines. Not sure if we should. Probably? There may be a bug here, too.